### PR TITLE
app-misc/openrgb: add 0.8

### PR DESCRIPTION
**Pull request by scardracs**
New version requires dev-ccp/cpp-httplib which is on ::guru, so re are gonna move it into ::gentoo.
It also requires mdns, so we are gonna add it too.

Closes: https://bugs.gentoo.org/886486

Signed-off-by: Marco Scardovi <mscardovi@icloud.com>

### CODE REVIEW

File: app-misc/openrgb/openrgb-0.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+fi
+
+DESCRIPTION="Open source RGB lighting control"
+HOMEPAGE="https://openrgb.org https://gitlab.com/CalcProgrammer1/OpenRGB/"
+LICENSE="GPL-2"
+# subslot is OPENRGB_PLUGIN_API_VERSION from https://gitlab.com/CalcProgrammer1/OpenRGB/-/blob/master/OpenRGBPluginInterface.h
+SLOT="0/2"
+
+RDEPEND="
+	dev-cpp/cpp-httplib
```
Comment (DarthGandalf) [inline app-misc/openrgb/openrgb-0.8.ebuild:27]: It will still use the bundled one, despite requesting the system one?

**Comment by DarthGandalf** [comment]
The commits are out of order: at the time openrgb is added, httplib doesn't exist yet

Also please update the 9999 ebuild

File: app-misc/openrgb/openrgb-0.8.ebuild (hunk 1 above)
Comment (scardracs) [reply app-misc/openrgb/openrgb-0.8.ebuild:27]: It's not finished yet, I've to figure out how to unbundle them

**Comment by scardracs** [comment]
> The commits are out of order: at the time openrgb is added, httplib doesn't exist yet
> 
> 
> 
> Also please update the 9999 ebuild

I will. Anyway is live ebuild really needed or we can drop it?

**Comment by DarthGandalf** [comment]
> It's not finished yet

Ah ok. Please tell when it's ready to review

>  Anyway is live ebuild really needed or we can drop it?

It's not *really* needed, but why do you want to drop it? It makes it easier to preview new features/open bugs to upstream early.

File: app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+diff --git a/qt/OpenRGBNanoleafSettingsPage/OpenRGBNanoleafScanningThread.cpp b/qt/OpenRGBNanoleafSettingsPage/OpenRGBNanoleafScanningThread.cpp
+index 1e3c926b..27565b78 100644
+--- a/qt/OpenRGBNanoleafSettingsPage/OpenRGBNanoleafScanningThread.cpp
++++ b/qt/OpenRGBNanoleafSettingsPage/OpenRGBNanoleafScanningThread.cpp
+@@ -10,7 +10,7 @@
+ #include <ifaddrs.h>
+ #endif
+ 
+-#include "mdns.h"
++#include <mdns.h>
```
Comment (DarthGandalf) [inline app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch:44]: This shouldn't be necessary

File: app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ 
+ #include "OpenRGBNanoleafScanningThread.h"
+ 
+@@ -42,7 +42,7 @@ static mdns_string_t ipv4_address_to_string(char* buffer, size_t capacity, const
+ 	mdns_string_t str;
+ 	str.str = buffer;
+ 	str.length = len;
+-
```
Comment (DarthGandalf) [inline app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch:52]: neither is this

File: app-misc/openrgb/openrgb-0.8.ebuild [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+"
+
+PATCHES+=(
+	"${FILESDIR}"/OpenRGB-0.7-r1-udev.patch
+	"${FILESDIR}"/OpenRGB-0.8-drop-bundled-libs.patch
+)
+
+src_prepare() {
+	default
+	rm -r dependencies/{hidapi,libusb,json,mbedtls}* || die "Failed to remove unneded deps"
```
Comment (DarthGandalf) [inline app-misc/openrgb/openrgb-0.8.ebuild:50]: add the new to libs to the list

File: app-misc/openrgb/openrgb-0.8.ebuild [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	dodoc README.md OpenRGB.patch
+	udev_dorules 60-openrgb.rules
+
+	# This is for plugins. Upstream doesn't install any headers at all.
+	insinto /usr/include/OpenRGB
+	find . -name '*.h' -exec cp --parents '{}' "${ED}/usr/include/OpenRGB/" ';' || die
+}
+
+pkg_postinst() {
+	xdg_icon_cache_update
```
Comment (DarthGandalf) [inline app-misc/openrgb/openrgb-0.8.ebuild:76]: this needs xdg-utils eclass

does it even work now without that eclass?

File: app-misc/openrgb/openrgb-0.8.ebuild [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+fi
+
+DESCRIPTION="Open source RGB lighting control"
+HOMEPAGE="https://openrgb.org https://gitlab.com/CalcProgrammer1/OpenRGB/"
+LICENSE="GPL-2"
+# subslot is OPENRGB_PLUGIN_API_VERSION from https://gitlab.com/CalcProgrammer1/OpenRGB/-/blob/master/OpenRGBPluginInterface.h
+SLOT="0/2"
+
+RDEPEND="
+	dev-cpp/cpp-httplib
```
Comment (DarthGandalf) [inline app-misc/openrgb/openrgb-0.8.ebuild:25]: If this is a header-only library, it's not actually needed in runtime. So move it from RDEPEND to DEPEND

File: app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+     dependencies/libcmmk/include/                                                               \
+-    dependencies/mdns                                                                           \
+     i2c_smbus/                                                                                  \
+     i2c_tools/                                                                                  \
+     net_port/                                                                                   \
+@@ -1562,6 +1560,7 @@ contains(QMAKE_PLATFORM, linux) {
+     -lmbedx509                                                                                  \
+     -lmbedtls                                                                                   \
+     -lmbedcrypto                                                                                \
++    -httplib
```
Comment (DarthGandalf) [inline app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch:31]: How does this work?

File: app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch (hunk 2 above)
Comment (scardracs) [reply app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch:44]: How do you call /usr/include/mdns.h without an #include?

File: app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch (hunk 7 above)
Comment (scardracs) [reply app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch:31]: It does as I can see

File: app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch (hunk 2 above)
Comment (DarthGandalf) [reply app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch:44]: Not without. `#include "mdns.h"` works as well. The patch is not needed.

Comment (scardracs) [reply app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch:50]: I've tried and fails every single time.. It works only in that way

Comment (scardracs) [reply app-misc/openrgb/files/OpenRGB-0.8-drop-bundled-libs.patch:50]: When you use "" it checks on libdirs you have specified before while <> tell it to look for system's libraries placed inside /usr/include

File: app-misc/openrgb/openrgb-9999.ebuild [hunk 8]
```diff
@@ -12,19 +12,18 @@ else
...
-DESCRIPTION="Open source RGB lighting control that doesn't depend on manufacturer software"
+DESCRIPTION="Open source RGB lighting control"
 HOMEPAGE="https://openrgb.org https://gitlab.com/CalcProgrammer1/OpenRGB/"
 LICENSE="GPL-2"
 # subslot is OPENRGB_PLUGIN_API_VERSION from https://gitlab.com/CalcProgrammer1/OpenRGB/-/blob/master/OpenRGBPluginInterface.h
 SLOT="0/2"
 
 RDEPEND="
 	dev-libs/hidapi
+	dev-libs/mdns
```
Comment (DarthGandalf) [inline app-misc/openrgb/openrgb-9999.ebuild:26]: Isn't this also header-only?

**Comment by scardracs** [comment]
@DarthGandalf is it ok to you if I add you as maintainer instead of me for these libs? If not I'm gonna add m-n

**Pull request closed without merging**